packages/lib/gpc/src/gpcChecks.ts

```diff
@@ -538,24 +522,114 @@ export function checkVerifyClaimsForConfig(
       }
     }
   }
+  // TODO(POD-P2): List membership claims
+}
+
+/**
+ * Picks the smallest available circuit in this family which can handle the
+ * size parameters of a desired configuration.
+ *
+ * @param circuitReq the circuit size requirements
+ * @returns the circuit description, or undefined if no circuit can handle
+ *   the required parameters.
+ * @throws Error if there are no circuits satisfying the given requirements.
+ */
+export function circuitDescForRequirements(
+  circuitReq: GPCRequirements
+): ProtoPODGPCCircuitDesc {
+  for (const circuitDesc of ProtoPODGPC.CIRCUIT_FAMILY) {
+    if (circuitDescMeetsRequirements(circuitDesc, circuitReq)) {
+      return circuitDesc;
+    }
+  }
+
+  throw new Error(
+    `There are no circuit parameters satisfying these requirements: ${circuitReq}`
+  );
+}
+
+/**
+ * Checks whether a described circuit can meet given GPC size requirements.
+ * This will be true if each of the circuit's parameters is greater than or
+ * equal to the required value.
+ *
+ * @param circuitDesc description of the circuit to check
+ * @param circuitReq the circuit size requirements
+ * @returns `true` if the circuit meets the requirements.
+ */
+export function circuitDescMeetsRequirements(
+  circuitDesc: ProtoPODGPCCircuitDesc,
+  circuitReq: GPCRequirements
+): boolean {
+  // Check tuple parameter compatibility.
+  // Indicate negative tuple requirement check if `requiredNumTuples` throws due
+  // to an invalid choice of parameters (i.e. `params.tupleArity < 2`).
+  const tupleCheck = ((): boolean => {
+    try {
+      // The circuit description should have enough tuples of arity `tupleArity` to
+      // cover all input tuples.
+      return (
+        circuitDesc.maxTuples >=
+        circuitReq.tupleArities
+          .map((arity) => requiredNumTuples(circuitDesc.tupleArity, arity))
+          .reduce((sum, requiredNum) => sum + requiredNum, 0)
+      );
+    } catch (_) {
```

Comment:
I don't broad catches like this, since you can catch things you don't intend to.  You want to sparate unmet requirements from invalid input.

If tupleArity < 2 is the only requirement you intend to be checking here, maybe you should check it explicitly?  Or have a version of requiredNumTuples which signals unmeetable requirements with a return value.  Or throw a specific type of named exception which can be checked here, but that might be overkill complexity.

Reply:
I don't love the fragility of relying on error message text in the latest version of this, but I don't think it's a big enough deal to worry about right now.  TypeScript doesn't make distinguishing exceptions as easy as I'm used to from languages like C++ or Java.  For our network API layers I introduced a NamedError helper for that, but it would probably be overkill for a case like this.